src/components/progress-circle/progress_circle.spec.ts

```diff
@@ -0,0 +1,91 @@
+
+    beforeEach(inject([TestComponentBuilder], (tcb:TestComponentBuilder) => {
+      builder = tcb;
+    }));
+
+    it('applies a mode of "determinate" if no mode is provided.', (done:() => void) => {
+      builder
+        .overrideTemplate(TestApp, '<md-progress-circle></md-progress-circle>')
+        .createAsync(TestApp)
+        .then((fixture) => {
+          let progressElement = getChildDebugElement(fixture.debugElement, 'md-progress-circle');
+          expect(progressElement.componentInstance.mode).toBe('determinate');
+          done();
+        });
+    });
+
+    it('applies a mode of "determinate" if an invalid mode is provided.', (done:() => void) => {
+      builder
+        .overrideTemplate(TestApp, '<md-progress-circle mode="spinny"></md-progress-circle>')
+        .createAsync(TestApp)
+        .then((fixture) => {
+          let progressElement = getChildDebugElement(fixture.debugElement, 'md-progress-circle');
+          expect(progressElement.componentInstance.mode).toBe('determinate');
+          done();
+        });
+    });
+
+    it('does not modify the mode if a valid mode is provided.', (done:() => void) => {
+      builder
+        .overrideTemplate(TestApp, '<md-progress-circle mode="indeterminate"></md-progress-circle>')
+        .createAsync(TestApp)
+        .then((fixture) => {
+          let progressElement = getChildDebugElement(fixture.debugElement, 'md-progress-circle');
+          expect(progressElement.componentInstance.mode).toBe('indeterminate');
+          done();
+        });
+    });
+
+    it('defines a default value for the value attribute', (done:() => void) => {
+      builder
+        .overrideTemplate(TestApp, '<md-progress-circle></md-progress-circle>')
+        .createAsync(TestApp)
+        .then((fixture) => {
+          let progressElement = getChildDebugElement(fixture.debugElement, 'md-progress-circle');
+          expect(progressElement.componentInstance.value).toBe(0);
+          done();
+        });
+    });
+
+    it('clamps the value of the progress between 0 and 100', (done:() => void) => {
+      builder
+        .overrideTemplate(TestApp, '<md-progress-circle></md-progress-circle>')
+        .createAsync(TestApp)
+        .then((fixture) => {
+          let progressElement = getChildDebugElement(fixture.debugElement, 'md-progress-circle');
+          let progressComponent = progressElement.componentInstance;
+
+          progressComponent.value = 50;
+          expect(progressComponent.value).toBe(50);
+
+          progressComponent.value = 999;
+          expect(progressComponent.value).toBe(100);
+
+          progressComponent.value = -10;
+          expect(progressComponent.value).toBe(0);
+          done();
+        });
+    });
+  });
+}
+
+
+/** Gets a child DebugElement by tag name. */
+function getChildDebugElement(parent: DebugElement, tagName: string): DebugElement {
+  return parent.query(debugEl => debugEl.nativeElement.tagName.toLowerCase() == tagName);
+}
```

Comment:
You shouldn't need this function any more; you can instead:
```
import {By} from 'angular2/platform/browser';
...
fixture.debugElement.query(By.css('selector'))
```
(see the button spec)